pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFieldSpec.java

```diff
@@ -109,7 +113,9 @@ public String getGranularity() {
 
   // Required by JSON de-serializer. DO NOT REMOVE.
   public void setGranularity(@Nonnull String granularity) {
+    Preconditions.checkArgument(DateTimeGranularitySpec.isValidGranularity(granularity));
     _granularity = granularity;
+    _granularitySpec = new DateTimeGranularitySpec(granularity);
```

Comment:
Seems like JSON serde should work with this change, but would be good to ensure via a unit test.